test/ConnectTest.php

```diff
@@ -250,4 +250,50 @@ public function testConnectWithUri()
             $this->assertContains('Invalid credentials', $zle->getMessage());
         }
     }
+
+    /**
+     * This test checks whether the default ports are set correctly when no port
+     * is given in the configuration.
+     *
+     * This has been reported with issue 19 of Zend\Ldap
+     *
+     * @param string $host       The host to connect to
+     * @param bool   $ssl        Whether to use ssl or not
+     * @param strnig $connectUri The expected connectionURI
+     *
+     * @see https://github.com/zendframework/zend-ldap/issues/19
+     * @dataProvider connectionWithoutPortInOptionsArrayProvider
+     */
+    public function testConnectionWithoutPortInOptionsArray($host, $ssl, $connectURI)
+    {
+        $options = [
+            'host' => $host,
+            'useSsl' => $ssl,
+        ];
+
+        try {
+            $ldap = new \Zend\Ldap\Ldap($options);
+            $ldap->connect();
+
+            $r = new \ReflectionObject($ldap);
+            $p = $r->getProperty('connectString');
+            $p->setAccessible(true);
+            $this->assertEquals(
+                $connectURI,
+                $p->getValue($ldap)
+            );
+        } catch (Exception\LdapException $e) {
+            $this->fail($e->getMessage());
```

Comment:
There is no reason for catch the exception and then mark it as fail. PHPUnit already catch the exceptions and mark the test as failure